platform-api/src/main/java/io/datacater/core/deployment/DeploymentEndpoint.java

```diff
@@ -161,36 +163,63 @@ public Uni<List<DeploymentEntity>> getDeployments() {
   @Consumes(MediaType.APPLICATION_JSON)
   public Uni<DeploymentEntity> createDeployment(DeploymentSpec spec) {
     DeploymentEntity de = new DeploymentEntity(spec);
-    Uni<PipelineEntity> pipelineUni = getPipeline(spec);
+    Uni<List<ConfigEntity>> configList =
+        ConfigUtilities.getConfig(ConfigUtilities.getConfigNames(spec.labels()), dsf);
+
     return dsf.withTransaction(
         (session, transaction) ->
             session
                 .persist(de)
                 .onItem()
-                .transformToUni(empty -> pipelineUni)
+                .transform(
+                    x ->
+                        configList
+                            .onItem()
+                            .transform(
+                                configs -> ConfigUtilities.combineWithDeployment(spec, configs)))
                 .onItem()
                 .transformToUni(
-                    pipelineEntity ->
+                    combinedSpecUni ->
```

Comment:
I feel that we should try to keep the endpoints as flat as possible and move the application of configs to deployments and streams out of here.

The `DeploymentEndpoint` should not know about configs. That should be implemented in, for instance, the [K8Deployment](https://github.com/DataCater/datacater/blob/main/platform-api/src/main/java/io/datacater/core/deployment/K8Deployment.java) class.

Reply:
yes, i do feel the same way. The problem here is that when we retrieve/work with entities with mutiny, it all has to happen inside the worker thread that is returned to the user for execution. If we were to move the config retrieval from here into the K8Deployment class, it would cause a more complicated/distributed nesting making it a lot less readable and harder to work with in my opinion.

Reply:
created issue #174